tests/testsuite/rustflags.rs

```diff
@@ -1478,36 +1481,44 @@ fn env_rustflags_misspelled_build_script() {
 
 #[cargo_test]
 fn remap_path_prefix_ignored() {
-    // Ensure that --remap-path-prefix does not affect metadata hash.
-    let p = project().file("src/lib.rs", "").build();
-    p.cargo("build").run();
-    let rlibs = p
-        .glob("target/debug/deps/*.rlib")
-        .collect::<Result<Vec<_>, _>>()
-        .unwrap();
-    assert_eq!(rlibs.len(), 1);
-    p.cargo("clean").run();
-
-    let check_metadata_same = || {
-        let rlibs2 = p
-            .glob("target/debug/deps/*.rlib")
-            .collect::<Result<Vec<_>, _>>()
-            .unwrap();
-        assert_eq!(rlibs, rlibs2);
+    let get_c_metadata_re =
+        regex::Regex::new(r".*(--crate-name [^ ]*).*(-C ?metadata=[^ ]*).*").unwrap();
+    let get_c_metadata = |output: RawOutput| {
+        let stderr = String::from_utf8(output.stderr).unwrap();
+        let mut c_metadata = get_c_metadata_re
+            .captures_iter(&stderr)
+            .map(|c| {
+                let (_, [name, c_metadata]) = c.extract();
+                format!("{name} {c_metadata}")
+            })
+            .collect::<Vec<_>>();
+        assert!(
+            !c_metadata.is_empty(),
+            "`{get_c_metadata_re:?}` did not match:\n```\n{stderr}\n```"
+        );
+        c_metadata.sort();
+        c_metadata.join("\n")
     };
 
-    p.cargo("build")
+    let p = project().file("src/lib.rs", "").build();
+
+    let build_output = p
+        .cargo("build -v")
         .env(
             "RUSTFLAGS",
             "--remap-path-prefix=/abc=/zoo --remap-path-prefix /spaced=/zoo",
         )
         .run();
-    check_metadata_same();
+    let build_c_metadata = dbg!(get_c_metadata(build_output));
```

Comment:
```suggestion
    let build_c_metadata = get_c_metadata(build_output);
```

Reply:
I intentionally left `dbg` statements in so people can vouch for the output thats being asserted since the assertion's expected result is another command-under-test which is questionable.  The very first time I ran this test, it passed.  The output was blank.  I added the assert for that reason but figured it would still be good to be able to verify it.

(lots of other tests print results as they go)